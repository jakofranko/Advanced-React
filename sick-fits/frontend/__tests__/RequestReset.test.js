import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import Router from 'next/router';
import RequestReset, { REQUESTRESET_MUTATION } from '../components/RequestReset';
import { MockedProvider } from 'react-apollo/test-utils';

const mocks = [{
  request: {
    query: REQUESTRESET_MUTATION,
    variables: { email: 'unit@test.com' }
  },
  result: {
    data: {
      requestReset: {
        message: 'success',
        __typename: 'Message'
      }
    }
  }
}];

describe('<RequestReset />', () => {
  it('renders and matches snapshot', async () => {
    const comp = mount(
      <MockedProvider>
        <RequestReset />
      </MockedProvider>
    );
    const form = comp.find('[data-test="RequestRest"]');
    expect(toJSON(form)).toMatchSnapshot();
  });

  it('calls the mutation', async () => {
    const comp = mount(
      <MockedProvider mocks={mocks}>
        <RequestReset />
      </MockedProvider>
    );

    comp
      .find('input')
      .simulate(
        'change',
        {
          target: {
            name: 'email',
            value: 'unit@test.com'
          }
        }
      );
    comp.find('form').simulate('submit');

    await wait();
    comp.update();

    expect(comp.find('p').text()).toContain('Success! Check you email for a reset link');
  });
});
