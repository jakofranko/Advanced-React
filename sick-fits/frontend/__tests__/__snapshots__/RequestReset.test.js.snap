// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset /> renders and matches snapshot 1`] = `
Array [
  <Form
    data-test="RequestRest"
    method="post"
    onSubmit={[Function]}
  >
    <form
      className="Form-sc-1xszr8q-0 krRxky"
      data-test="RequestRest"
      method="post"
      onSubmit={[Function]}
    >
      <fieldset
        aria-busy={false}
        disabled={false}
      >
        <h2>
          Request a Password Reset
        </h2>
        <DisplayError
          error={Object {}}
        />
        <label
          htmlFor="email"
        >
          Email
          <input
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="email"
            value=""
          />
        </label>
        <button
          type="submit"
        >
          Reset Password
        </button>
      </fieldset>
    </form>
  </Form>,
  <form
    className="Form-sc-1xszr8q-0 krRxky"
    data-test="RequestRest"
    method="post"
    onSubmit={[Function]}
  >
    <fieldset
      aria-busy={false}
      disabled={false}
    >
      <h2>
        Request a Password Reset
      </h2>
      <DisplayError
        error={Object {}}
      />
      <label
        htmlFor="email"
      >
        Email
        <input
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="email"
          value=""
        />
      </label>
      <button
        type="submit"
      >
        Reset Password
      </button>
    </fieldset>
  </form>,
]
`;
